### PR TITLE
trigger/resource: support namespace limiting

### LINKED ISSUE #16: feature request: allow namespace allow-listing for resource trigger
Currently, the `resource` trigger sets up a client watching all namespaces for a configured resource, compare https://github.com/brancz/locutus/blob/2b6852bd3b57312c103d1d2466a23afef70f8ebb/trigger/resource/resource.go#L80

Does it make sense to introduce a `namespace` configuration for the `Resource` trigger type (or even an allow-list `namespaces`) to scope down that client?

UX suggestion:
Add a `namespace` stanza to the resource trigger configuration:
```
--- a/trigger/resource/resource.go
+++ b/trigger/resource/resource.go
@@ -38,6 +38,7 @@ type ResourceTriggerConfig struct {
        Name                     string                    `json:"name"`
        Kind                     string                    `json:"kind"`
        APIVersion               string                    `json:"apiVersion"`
+       Namespace               string                  `json:"namespace"`
        KeyTransformationConfigs []KeyTransformationConfig `json:"keyTransformations"`
 }
```

cc @paulfantom @metalmatze

### PR DESCRIPTION
**Pull request by s-urbaniak**
Currently a trigger resource sets up a client for all namespaces.
This adds the capability to limit a resource trigger config to an allow-listed namespace.

Signed-off-by: Sergiusz Urbaniak <sergiusz.urbaniak@gmail.com>

Fixes #16

### CODE REVIEW

**Comment by s-urbaniak** [comment]
I just figured out that there is a `mainResource` setting which is incompatible with this change, hence this doesn't work yet, let me play around.

**Comment by s-urbaniak** [comment]
ok, this seems to be the easiest path forward for now, curious if this is feasible :-)

**Comment by brancz** [comment]
Love it!

lgtm

File: trigger/resource/resource.go [hunk 1]
```diff
@@ -77,7 +78,12 @@ func NewTrigger(logger log.Logger, client *client.Client, configFile string) (*T
 	}
 
 	for _, r := range config.Resources {
-		c, err := client.ClientFor(r.APIVersion, r.Kind, "")
+		key := r.Name
+		if len(r.Namespace) > 0 {
+			key = r.Namespace + "/" + r.Name
```
Comment (brancz) [inline trigger/resource/resource.go:83]: Actually on second thought, I don't think this is needed. The user is responsible for making sure these don't clash, `r.Name` is just any name that the user wants to give this resource-source.

Do you mind reverting this part?

We should just check that `t.infs[key]` doesn't already exist when we write it to that map.

Comment (s-urbaniak) [reply trigger/resource/resource.go:83]: ahh, yes this makes sense, so the name is completely user-defined. yes, let me revert it in a follow-up.

**Pull request merged**